docs/development/getting-started-locally.md

```diff
@@ -0,0 +1,162 @@
+
+    Replace the dummy values with the actual configurations and make sure to add a name and a namespace to the secret as intended.
+
+    > **Note 1:** The secret should be applied in the same namespace as druid.
+    >
+    > **Note 2:** All the values in the data field of secret yaml should be in base64 encoded format.
+
+  - **Apply the secret**
+
+    ```sh
+    kubectl apply -f path/to/secret
+    ```
+
+  - **Update Etcd**
+
+    Uncomment the `spec.backup.store` section of the druid yaml and set the keys to allow backuprestore to take backups by connecting to an object store.
+
+    ```yaml
+    # Configuration for storage provider
+    store:
+        secretRef:
+            name: etcd-backup-secret-name
+        container: object-storage-container-name
+        provider: aws # options: aws,azure,gcp,openstack,alicloud,dell,openshift,local
+        prefix: etcd-test
+    ```
+
+    Brief explanation of keys:
+
+    - `secretRef.name` is the name of the secret that was applied as mentioned above
+    - `store.container` is the object storage bucket name
+    - `store.provider`  is the bucket provider. Pick from the options mentioned in comment
+    - `store.prefix`    is the folder name that you want to use for your snapshots inside the bucket.
+
+### Applying the Etcd CR
+
+> **Note:** With backups enabled, make sure the bucket is created in corresponding cloud provider before applying the Etcd yaml
+
+Create the Etcd CR (Custom Resource) by applying the Etcd yaml to the cluster
+
+```sh
+# Apply the prepared etcd CR yaml
+kubectl apply -f config/samples/druid_v1alpha1_etcd.yaml
+```
+
+### Annotate Etcd CR with the reconcile annotation
+
+> **Note :** If the `ignore-operation-annotation` flag is set to `true`, this step is not required
+
+The above step creates an Etcd resource, however etcd-druid won't pick it up for reconciliation without an annotation. To get etcd-druid to reconcile the etcd CR, annotate it with the following `gardener.cloud/operation:reconcile`.
+
+```sh
+# Annotate etcd-test CR to reconcile
+kubectl annotate etcd etcd-test gardener.cloud/operation="reconcile"
+```
+
+This starts creating the etcd cluster
+
+### Verify the Etcd cluster
+
+To obtain information regarding the newly instantiated etcd cluster, perform the following step, which gives details such as the cluster size, readiness status of its members, and various other attributes.
+
+```sh
+kubectl get etcd -o=wide
+```
+
+#### Verify Etcd Member Pods
+
+To check the etcd member pods, do the following and look out for pods starting with the name `etcd-`
+
+```sh
+kubectl get pods
+```
+
+#### Verify Etcd Pods' Functionality
+
+Verify the working conditions of the etcd pods by putting data through a etcd container and access the db from same/another container depending on single/multi node etcd cluster
+
+```sh
+# Put a key-value pair into the etcd 
+kubectl exec -it <etcd_pod> -c etcd -- etcdctl put key1 value1
```

Comment:
We cannot assume that `exec` will work, specifically for clusters using etcd-wrapper. Instead, You can add a line before this to either exec into the etcd container with a bash session using `kubectl exec -it ... -- bash`, or start an ephemeral container and attach it to the etcd container using `kubectl debug ...` (you may also link to [this doc](https://github.com/gardener/etcd-wrapper/blob/main/docs/deployment/ops.md#using-the-image)). Once that's done, you can simply specify the `etcdctl` commands directly.

Reply:
Thanks for pointing it out @shreyas-s-rao. Updated in the latest commit.